release/internal/pinnedversion/pinnedversion.go

```diff
@@ -286,19 +299,37 @@ func LoadHashrelease(repoRootDir, outputDir, hashreleaseSrcBaseDir string, lates
 	}, nil
 }
 
+// RetrieveImageComponents retrieves the images from Calico components in the pinned version file that produce images.
+// It also adds the Tigera operator and its init image to the returned map.
+//
+// Images returned from this function are expected to be in the format "<registry>/<image-name>" where <registry> includes the registry and image path. However, the versions.yml file represents images differently, including the image path as the first part of the image name instead of as part of the registry.
+// As a result, the image path is stripped from the image name to only return the image name so that images are properly formatted.
+//
+// For example, if the image name is "calico/node", this allows the fully qualified image to be "quay.io/calico/node" when a registry of `quay.io/calico` is prepended.
+// and prevents duplication of the image path(i.e. "quay.io/calico/calico/node").
 func RetrieveImageComponents(outputDir string) (map[string]registry.Component, error) {
 	pinnedVersion, err := retrievePinnedVersion(outputDir)
 	if err != nil {
 		return nil, err
 	}
 	components := pinnedVersion.Components
 	for name, component := range components {
-		// Remove components that do not produce images.
-		if utils.Contains(excludedComponents, name) {
+		// Remove components that should be excluded. Either because they do not have an image, or not built by Calico.
+		if slices.Contains(noImageComponents, name) || name == flannelComponentName {
 			delete(components, name)
 			continue
 		}
-		components[name] = normalizeComponent(name, component)
+		img := registry.ImageMap[name]
+		if img == "" {
+			img = name
+		}
+		component.Image = img
```

Comment:
Just noticed, but this phrasing is a bit awkward. 

We assign a local variable `img`, and halfway through processing the name we switch to operating on `component.Image`

I'd suggest either:
- doing all the processing on `img` and assigning the value to `component.Image` only at the very end.
- getting rid of the `img` intermediate, and assigning to `component.Image` throughout.